laplace/baselaplace.py

```diff
@@ -1821,8 +1909,1041 @@ def sample(
+
+    def __call__(
+        self,
+        x: torch.Tensor | MutableMapping,
+        pred_type: PredType | str = PredType.GP,
+        joint: bool = False,
+        link_approx: LinkApprox | str = LinkApprox.PROBIT,
+        n_samples: int = 100,
+        diagonal_output: bool = False,
+        generator: torch.Generator | None = None,
+        fitting: bool = False,
+        **model_kwargs: dict[str, Any],
+    ) -> torch.Tensor | tuple[torch.Tensor, torch.Tensor]:
+        """Compute the posterior predictive on input data `x`.
+
+        Parameters
+        ----------
+        x : torch.Tensor or MutableMapping
+            `(batch_size, input_shape)` if tensor. If MutableMapping, must contain
+            the said tensor.
+
+        pred_type : {'gp'}, default='gp'
+            type of posterior predictive, linearized GLM predictive (GP).
+            The GP predictive is consistent with
+            the curvature approximations used here.
+
+        link_approx : {'mc', 'probit', 'bridge', 'bridge_norm'}
+            how to approximate the classification link function for the `'glm'`.
+
+        joint : bool
+            Whether to output a joint predictive distribution in regression with
+            `pred_type='glm'`. If set to `True`, the predictive distribution
+            has the same form as GP posterior, i.e. N([f(x1), ...,f(xm)], Cov[f(x1), ..., f(xm)]).
+            If `False`, then only outputs the marginal predictive distribution.
+            Only available for regression and GLM predictive.
+
+        n_samples : int
+            number of samples for `link_approx='mc'`.
+
+        diagonal_output : bool
+            whether to use a diagonalized posterior predictive on the outputs.
+            Only works for `link_approx='mc'`.
+
+        generator : torch.Generator, optional
+            random number generator to control the samples (if sampling used).
+
+        fitting : bool, default=False
+            whether or not this predictive call is done during fitting. Only useful for
+            reward modeling: the likelihood is set to `"regression"` when `False` and
+            `"classification"` when `True`.
+
+        Returns
+        -------
+        predictive: torch.Tensor or Tuple[torch.Tensor]
+            For `likelihood='classification'`, a torch.Tensor is returned with
+            a distribution over classes (similar to a Softmax).
+            For `likelihood='regression'`, a tuple of torch.Tensor is returned
+            with the mean and the predictive variance.
+            For `likelihood='regression'` and `joint=True`, a tuple of torch.Tensor
+            is returned with the mean and the predictive covariance.
+        """
+        if self._fitted is False:
+            raise RuntimeError(
+                "Functional Laplace has not been fitted to any "
+                + "training dataset. Please call .fit method."
+            )
+
+        if self._recompute_Sigma is True:
+            Warning(
```

Comment:
Before, you use `warnings.warn` and here just `Warning`. What is the difference? Please make it consistent. According to ruff, it should be `raise Warning` by the way.